fabric-item-api-v1/src/main/java/net/fabricmc/fabric/impl/item/FabricItemInternals.java

```diff
@@ -16,42 +16,35 @@
 
 package net.fabricmc.fabric.impl.item;
 
+import java.util.Collections;
+import java.util.Map;
 import java.util.WeakHashMap;
 
 import net.minecraft.item.Item;
 
+import net.fabricmc.fabric.api.item.v1.CustomItemSettingType;
 import net.fabricmc.fabric.api.item.v1.CustomDamageHandler;
 import net.fabricmc.fabric.api.item.v1.EquipmentSlotProvider;
 
 public final class FabricItemInternals {
-	private static final WeakHashMap<Item.Settings, ExtraData> extraData = new WeakHashMap<>();
+	public static final CustomItemSettingType<EquipmentSlotProvider> EQUIPMENT_SLOT_PROVIDER = CustomItemSettingType.of(() -> null);
+	public static final CustomItemSettingType<CustomDamageHandler> CUSTOM_DAMAGE_HANDLER = CustomItemSettingType.of(() -> null);
+
+	private static final Map<Item.Settings, Map<CustomItemSettingType<?>, Object>> customSettings = new WeakHashMap<>();
```

Comment:
Why not just add a field to `Item.Settings` though mixin?

Reply:
That would be my first approach as well. This is how it was done [prior to this PR](https://github.com/FabricMC/fabric/blob/616c01224d0b1009f7a635311f87ce2f3bb50de0/fabric-item-api-v1/src/main/java/net/fabricmc/fabric/impl/item/FabricItemInternals.java) (with `ExtraData` instead of `Map<CustomItemSettingType<?>, Object>`, but still) so I chose to stick with that.

Reply:
Imo you should follow what block settings has https://github.com/FabricMC/fabric/blob/1e6c71f9bcb3fe386d9fa8030d061b37b9afe31e/fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/mixin/object/builder/AbstractBlockSettingsMixin.java#L30

Also would be good to add this to block settings

Reply:
I've changed it to follow this. And I agree that it would be good to add something similar to block settings, but as that's a different module, it would be a different PR.